pkg/agent/agent_test.go

```diff
@@ -134,3 +141,155 @@ func TestGetRoundInfo(t *testing.T) {
+	ipDevice := &net.Interface{
+		Index:        10,
+		MTU:          1500,
+		Name:         "ens160",
+		HardwareAddr: macAddr,
+	}
+	podCIDRStr := "172.16.10.0/24"
+	_, podCIDR, _ := net.ParseCIDR(podCIDRStr)
+	tests := []struct {
+		name                   string
+		trafficEncapMode       config.TrafficEncapModeType
+		tunnelType             ovsconfig.TunnelType
+		mtu                    int
+		expectedNodeConfig     config.NodeConfig
+		expectedNodeAnnotation map[string]string
+	}{
+		{
+			name:             "noencap mode",
+			trafficEncapMode: config.TrafficEncapModeNoEncap,
+			mtu:              0,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1500,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: map[string]string{types.NodeMACAddressAnnotationKey: macAddr.String()},
+		},
+		{
+			name:             "hybrid mode",
+			trafficEncapMode: config.TrafficEncapModeHybrid,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1500,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: map[string]string{types.NodeMACAddressAnnotationKey: macAddr.String()},
+		},
+		{
+			name:             "encap mode",
+			trafficEncapMode: config.TrafficEncapModeEncap,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1500,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: nil,
+		},
+		{
+			name:             "encap mode, geneve tunnel",
+			trafficEncapMode: config.TrafficEncapModeEncap,
+			tunnelType:       ovsconfig.GeneveTunnel,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1450,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: nil,
+		},
+		{
+			name:             "encap mode, mtu specified",
+			trafficEncapMode: config.TrafficEncapModeEncap,
+			tunnelType:       ovsconfig.GeneveTunnel,
+			mtu:              1400,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1400,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
```

Comment:
nit: it seems that you could have a `getExpectedNodeConfig` helper function with the `expectedMTU` as a parameter to avoid repeating this? Don't feel strongly about it though.

Reply:
makes sense, changed to construct an expectedNodeConfig directly in `t.Run`